static/app/components/draggableTabs/draggableTab.tsx

```diff
@@ -73,28 +112,80 @@ function BaseTab(
     [to, orientation]
   );
 
+  const {dropProps, isDropTarget} = useDroppableItem(
+    {
+      target: {type: 'item', key: item.key, dropPosition: 'on'},
+    },
+    dropState,
+    ref
+  );
+
+  function Draggable({children}) {
+    const {dragProps, dragButtonProps, isDragging} = useDrag({
+      getAllowedDropOperations: () => ['move'],
+      getItems() {
+        return [
+          {
+            tab: JSON.stringify({key: item.key, value: children}),
+          },
+        ];
+      },
+    });
+
+    const draggableRef = useRef(null);
+    const {buttonProps} = useButton(
+      {...dragButtonProps, elementType: 'div'},
+      draggableRef
+    );
+
+    return (
+      <div
+        {...mergeProps(dragProps, buttonProps)}
+        ref={draggableRef}
+        className={`draggable ${isDragging ? 'dragging' : ''}`}
+      >
+        {children}
+      </div>
+    );
+  }
+
   return (
-    <TabWrap
-      {...tabProps}
-      hidden={hidden}
-      selected={isSelected}
-      overflowing={overflowing}
-      ref={ref}
-    >
-      <InnerWrap>
-        <StyledInteractionStateLayer
-          orientation={orientation}
-          higherOpacity={isSelected}
+    <Fragment>
+      <TabDropIndicator
+        target={{type: 'item', key: item.key, dropPosition: 'before'}}
+        dropState={dropState}
+      />
+      <TabWrap
+        {...mergeProps(tabProps, dropProps)}
+        hidden={hidden}
+        selected={isSelected}
+        overflowing={overflowing}
+        ref={ref}
+        className={`option ${isFocusVisible ? 'focus-visible' : ''} ${
+          isDropTarget ? 'drop-target' : ''
+        }`}
```

Comment:
is there a plan for these classnames?

Reply:
Nope, this was copy pasted over from the react-aria guides. I'll delete them if/until they become useful